### PR TITLE
Ability to add Web Configuration properties, bug fixes to updates

### PR DESCRIPTION
**Pull request by kevinsea**
Adds support for setting the LogFileDirectory. Fixed type casting issues updating and comparing Web Configuration Properties. And added support for creating new Web Configuration Properties (currently only updates are supported). Changed code flow to do new site creation before update to reduce the duplication of code between update and create. Improved error handling when an update fails. Some general code cleanup  Sorry, this should have been multiple commits.

### CODE REVIEW

File: DSCResources/PSHOrg_cWebsite/PSHOrg_cWebsite.psm1 [hunk 1]
```diff
@@ -165,80 +234,54 @@ function Set-TargetResource
...
                     
-                    Try
-                    {
-                        $propObject = Get-WebConfigurationProperty -Filter $prop.CimInstanceProperties["Filter"].Value -PSPath $PSPath -Name $prop.CimInstanceProperties["Name"].Value | Select Value
-                    }
-                    Catch [System.IO.FileNotFoundException]
+                    if ($propObject -eq $null) # its a new property
                     {
-                        Write-Warning("Exception: $($_.Exception.Message)")
+                        Write-Verbose("Creating property '$filter': '$propName'.");
```
Comment (dlwyatt) [inline DSCResources/PSHOrg_cWebsite/PSHOrg_cWebsite.psm1]: Should there be an `$updateNotRequired = $false` line here?

**Comment by dlwyatt** [comment]
I haven't had time to do any testing yet, but at first glance, this looks great!

Now that Microsoft has open-sourced the DSC resource kit (and is taking community code), you may want to go straight to the source and just submit your PR to https://github.com/PowerShell/xWebAdministration instead.  It'll be a slow process to get all of our community forks merged, but eventually, the cWebAdministration module should be going away.

**Comment by kevinsea** [comment]
Good catch on $updateNotRequired = $false.

It would be great to get this merged as least as far as Powershell.org now.  I have also written a soon to be open source DSC generator that inspects your current 'golden image' machine and creates DSC for app pools, sites, virtual directories, web applications and web properties.  Today is relies on my fork. 

I was hoping to have it rely on the Powershell.org code (the MS code is so far behind, I don't know when they will catch up). I will commit that app in the next few days, hopefully that is of some use to people.

**Comment by kevinsea** [comment]
FWIW, you can find my IIS centric DSC Generator at https://github.com/kevinsea/dsc-generator

**Comment by kevinsea** [comment]
Hi, have you had a chance to further review this?

**Comment by smurawski** [comment]
Hey @kevinsea - Examples/Sample_cWebsite_LogFileAndProperties.psd1 should be just a ps1 file.  Other than that, the changes look sane.  Make that change and I'll try to get this merged.

**Comment by kevinsea** [comment]
Great.  I fixed the file extension.

Thanks @smurawski

**Comment by dlwyatt** [comment]
I keep losing track of these after the initial notification.  Just discovered this search option on GitHub, though:  https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+user%3APowerShellOrg+is%3Aopen+

I'll be adding that as a link in my favorites so I can check in from time to time to see what's still open and needs reviewing.

**Comment by tysonjhayes** [comment]
@dlwyatt that query is awesome. I am so using this. Thank you so much for pointing this out!

**Comment by kevinsea** [comment]
need anything more from me?

k

**Comment by smurawski** [comment]
Nope. I'll merge in the morning. Thanks.

**Pull request merged**